device/transport/amqp/src/amqp.ts

```diff
@@ -682,6 +686,14 @@ export class Amqp extends EventEmitter implements DeviceTransport {
       }
     }
 
+    if (options.hasOwnProperty('productInfo')) {
+      if (typeof(options.productInfo) === 'string') {
+        this._customUserAgentString = options.productInfo;
+      } else {
+        throw new errors.InvalidOperationError('productInfo must be set as a string');
+      }
+    }
+
     /*Codes_SRS_NODE_DEVICE_AMQP_13_001: [ The setOptions method shall save the options passed in. ]*/
     this._options = options;
```

Comment:
>this._options = options; [](start = 4, length = 24)

here it looks like we keep a reference to the options in the instance. given this, do we need the _customUserAgentString property? could we just pull it from the _options object directly line 285 (checking that this_options existnce of course?)